### PR TITLE
capture whole filename including dirs and add parsing tests

### PR DESCRIPTION
**Pull request by joehand**
It seems like we weren't capturing the subdirectories before (and they weren't in the tests). 

I added some basic parsing tests for GET requests too. We can extend those now pretty easily for other edge cases.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -33,15 +33,15 @@ function HyperdriveHttp (getArchive) {
...
       : segs.shift()
-    var filename = segs.shift()
+    var filename = segs.join('/')
     var op = 'get'
 
     try {
       // check if we are serving archive at root
       key = key.replace(/\.changes$/, '')
-      encoding.decode(key)
+      encoding.encode(Buffer.from(key, 'hex'))
```
Comment (juliangruber) [inline index.js]: isn't this what `dat-encoding` is already doing? https://github.com/juliangruber/dat-encoding/blob/master/index.js#L12

File: index.js [hunk 2]
```diff
@@ -113,7 +113,7 @@ function archiveResponse (datUrl, archive, req, res) {
 
 function onerror (status, res) {
   if (typeof status !== 'number') {
-    console.error(status)
+    // console.error(status)
```
Comment (juliangruber) [inline index.js]: can we remove this line all together then?

File: test/misc.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    var getUrl = filePath ? url.resolve(rootUrl, filePath) : rootUrl
+    request.get(getUrl).on('error', function () {
+      return
+    })
+  })
+})
+
+test.onFinish(function () {
+  server.close()
+})
```
Comment (juliangruber) [inline test/misc.js:83]: 👍

**Review by juliangruber — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (joehand) [reply index.js]: I thought so, but its not checking if its the right length. This is just a hacky way to make sure its a Dat key (if there is no key then key = filename). 

You think there is a better way to see if that first segment is a key or file?

File: index.js (hunk 2 above)
Comment (joehand) [reply index.js]: Oh ya this was a placeholder. We can remove it, any idea of better errors besides just giving a 404? Not sure if this is something we want to show the user though.

File: index.js (hunk 1 above)
Comment (joehand) [reply index.js]: We could just do key.length === 64.

File: index.js (hunk 2 above)
Comment (joehand) [reply index.js]: Removed. I'll add debug in another PR.

File: index.js (hunk 1 above)
Comment (juliangruber) [reply index.js]: `dat-encoding@3.0.1` will now throw on invalid key lenghts. I thought I had a test for this but my test key was `"too short"` which obviously isn't valid hex either, so now I test for both.

**Comment by joehand** [comment]
omg that took me forever. 🚢
